src/main/java/common/DriverBinariesSetter.java

```diff
@@ -1,40 +1,134 @@
+            case CHROME -> setBinaryByName(CHROME);
+            default -> throw new IllegalStateException("Unexpected value: " + browser);
         }
     }
 
-    private static void setBinaryByName(String name) {
+    private static void setBinaryByName(Drivers name) throws MalformedURLException {
         //Java 14 switch
         switch (name) {
-            case FIREFOX -> WebDriverManager.firefoxdriver().setup();
-            case EDGE -> WebDriverManager.edgedriver()
-                    .driverVersion(System.getProperty("edge.driver.version",
-                            StringUtils.EMPTY)).setup();
-            default -> WebDriverManager.chromedriver()
-                    .driverVersion(System.getProperty("chrome.driver.version",
-                            StringUtils.EMPTY)).setup();
+            case FIREFOX -> {
+                WebDriverManager.firefoxdriver().setup();
+                driver = new FirefoxDriver();
+            }
+            case EDGE -> {
+                WebDriverManager.edgedriver()
+                        .driverVersion(System.getProperty("edge.driver.version",
+                                StringUtils.EMPTY)).setup();
+                driver = new EdgeDriver();
+            }
+            case SAUCE_LAB_CHROME -> callSauceLabPropertiesChrome();
+            case SAUCE_LAB_FIREFOX -> callSauceLabPropertiesFirefox();
+            case SAUCE_LAB_EDGE -> callSauceLabPropertiesEdge();
+            default -> {
+                WebDriverManager.chromedriver()
+                        .driverVersion(System.getProperty("chrome.driver.version",
+                                StringUtils.EMPTY)).setup();
+                driver = new ChromeDriver();
+            }
+        }
+        WebDriverRunner.setWebDriver(driver);
+    }
+
+    private static void callSauceLabPropertiesChrome() throws MalformedURLException {
+        URL url = new URL(ConfigurationManager.getProperty("SAUCE_URL"));
+        DesiredCapabilities desiredCapabilities = new DesiredCapabilities();
+        ChromeOptions chromeOptions = new ChromeOptions();
+        chromeOptions.addArguments("--lang=ukr");
+        desiredCapabilities.setCapability(ChromeOptions.CAPABILITY, chromeOptions);
+        desiredCapabilities.setCapability("browserName", CHROME);
+        desiredCapabilities.setCapability("browser", ConfigurationManager.getProperty("SAUCE_LAB_CHROME_VERSION"));
+        desiredCapabilities.setCapability("platform", ConfigurationManager.getProperty("SAUCE_LAB_PLATFORM_VERSION"));
+        desiredCapabilities.setCapability("avoidProxy", true);
+        desiredCapabilities.setCapability("username", ConfigurationManager.getProperty("SAUCE_USERNAME"));
+        desiredCapabilities.setCapability("accessKey", ConfigurationManager.getProperty("SAUCE_ACCESS_KEY"));
+        driver = new RemoteWebDriver(url, desiredCapabilities);
+    }
+
+    private static void callSauceLabPropertiesFirefox() throws MalformedURLException {
```

Comment:
There is a code duplication in different callSauceLabProperties methods. Please consider extracting commons e.g. making field DesiredCapabilities. It's valid only in case you are not going to implement a real Factory